docs/docsite/rst/dev_guide/testing_units_modules.rst

```diff
@@ -159,7 +161,7 @@ simulating an API. As for 'six', the 'mock' python package is bundled with Ansib
 Ensuring failure cases are visible with mock objects
 ----------------------------------------------------
 
-Functions like module.fail_json() are normally expected to terminate execution. When you 
+Functions like ``module.fail_json()`` are normally expected to terminate execution. When you
```

Comment:
Informational note, (writing this down since I only sporadically deal with tests and so someone may want to remind me to help them document this) I believe that since we no longer support python-2.4 we can also catch SystemExit instead of mocking objects to prevent early termination.  that might be simpler code for many unittests:

Use a stdout fixture that redirects stdout to a StringIO and then assert that (1) SystemExit was raised and (2) that stdout received the json we expected:

``` python
import sys
from io import StringIO

@pytest.fixture
def stdout():
    stdout = StringIO()
    real_stdout = sys.stdout
    sys.stdout = stdout
    yield stdout

    sys.stdout = real_stdout

# am is another fixture for an AnsibleModule.  I think some existing tests in units/basic
# have that fixture already written but I'm not certain

def test_fail_json(stdout, am):
    with pytest.raises(SystemExit) as err:
        am.fail_json(msg="we are here")
    assert json.loads(stdout.getvalue()) == {"msg": "we are here", "failed": True, [...]}
```